Feature: Bash Updates

nJim opened the pull request.

Commit:
bash: Removed deprecated options from:
brew instalation and GNU utilities

q
.

Commit:
bash: added aliases changes

nJim commented on setup/homebrew.sh:
```diff
@@ -7,15 +7,16 @@ command -v xcode-select >/dev/null 2>&1 || {
 brew update
 brew upgrade
-brew prune
+# JFV:brew prune is ddepricated and now runs as part of cleanup
+# brew prune
```
This is going to be a hard habit for you to break. Everything is under version control, so there is no need to explain your changes within a file or to keep depricated code. Commented out code is pretty frowned upon now days as no one trusts its or plans to use it.

Instead, just make the change and commit the change with a git commit message. `git commit -m "Removing depricated brew prune as it now runs in cleanup."`

Of course it's too late to do that now. But in the future, small commits with meaningful messages are very helpful to understand the history of a file.

nJim commented on home/.bash_aliases:
```diff
@@ -1,17 +1,66 @@
+alias dt="cd ~/Desktop"
+alias p="cd ~/Projects"
+
+# JFV: Not sure what to do about colors, add this as a placeholder
```
No need to add initials. Each line in a file is tracked by git and your user is associated with that change. Git has a 'blame' report that shows who wrote or modified every line and when the change happened. You can jump back to old commits and get a sense of the history from there. 

So this is an old pattern. Embrace git. No need to leave your name in the files.

nJim commented on home/.aliases:
```diff
@@ -0,0 +1,101 @@
+alias ll="ls -l -G -h -p" 
+
+# List all files colorized in long format with short size buffers postpend /
+alias ll="ls -l ${colorflag} -h -p"
```
Is this overriding the above alias?

nJim commented on home/.aliases:
```diff
@@ -0,0 +1,101 @@
+
+# Always use color output for `ls` or JFV preference
+alias ls="command ls ${colorflag}"
+alias ls="ls -F"
```
Is this overriding the above alias?

nJim commented on home/.aliases:
```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env bash
```
What's the difference between this file and the bash_alias file? Seems like duplication of efforts.

nJim commented on setup/homebrew.sh:
```diff
@@ -32,15 +33,21 @@ brew install coreutils
 # https://www.gnu.org/software/findutils/
-brew install findutils --with-default-names
+# JFV: removed --with-default-names, option deprecated
+#   Must add PATH="/usr/local/opt/findutils/libexec/gnubin:$PATH"
```
Careful. You've made an assumption of the file location which could change from computer or computer. Instead of `/usr/local` I would suggest executing `brew --prefix`.

Also, this is a very prescriptive path. I believe homebrew creates a central place with symlinks to all executables. So you can simply add something like: `PATH=$(brew --prefix)/bin:$(brew --prefix)/sbin:$PATH`.

I didn't test the thing above. But something like that should work for any brew executable.

VomeroConsulting commented:
Jimmy,

Lot's of good feedback... I need to dig into the "color" usage more, just
tried to disable everything for now. Was planned to add back in for I was
thinking you like them and I'm not 100% sure I want them.

Also, was working with a local .aliases file I had, will need to feed it
back into a single .bash_aliases file. I will need to look further into how
you managed the "brew prefix" for it looks like it a good choice.

I will try to make smaller commits (maybe after one big one).

Dad

On Tue, May 5, 2020 at 1:47 PM Jim Vomero <notifications@github.com> wrote:

> *@nJim* commented on this pull request.
> ------------------------------
>
> In setup/homebrew.sh
> <https://github.com/nJim/dotfiles/pull/1#discussion_r420293968>:
>
> > @@ -32,15 +33,21 @@ brew install coreutils
>
>  # Install GNU find, locate, updatedb, and xargs.
>  # https://www.gnu.org/software/findutils/
> -brew <https://www.gnu.org/software/findutils/-brew> install findutils --with-default-names
> +# JFV: removed --with-default-names, option deprecated
> +#   Must add PATH="/usr/local/opt/findutils/libexec/gnubin:$PATH"
>
> Careful. You've made an assumption of the file location which could change
> from computer or computer. Instead of /usr/local I would suggest
> executing brew --prefix.
>
> Also, this is a very prescriptive path. I believe homebrew creates a
> central place with symlinks to all executables. So you can simply add
> something like: PATH=$(brew --prefix)/bin:$(brew --prefix)/sbin:$PATH
>
> I didn't test the thing above. But something like that should work for any
> brew executable.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/nJim/dotfiles/pull/1#pullrequestreview-406017892>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AOOGJ5XAAUWS3TTSZ3LX4ETRQBGJPANCNFSM4MZYH2HQ>
> .
>

VomeroConsulting committed:
Initial commit; not sure if .bashrc should be configured ths way but seems to work

VomeroConsulting committed:
Initial commit; bulk of file cp from mathiasbynens
needs some love; working on /usr/local/opt/grep/libexec/gnubin:/usr/local/opt/gnu-sed/libexec/gnubin:/usr/local/opt/findutils/libexec/gnubin:/usr/local/opt/coreutils/libexec/gnubin:/Users/jimv/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin when running bash from a bash shell

VomeroConsulting committed:
Did best to remove color from prompt while
leaving the option to add color back

VomeroConsulting committed:
Removed JFV from comments
Also removed colors for now but tried to make it easy to add in later

VomeroConsulting committed:
Initial release of these files used in .bash_profile

VomeroConsulting committed:
Removed JFV comments
Added VS Code and Python, have not yet tried these packages

VomeroConsulting committed:
Copied contents of osx.sh to macos.sh, files identical at this time.
Plan to modify such that JFV and use macos.sh for customization and
JLV can use osx.sh

VomeroConsulting committed:
Initial version of install script and required support file
dot-src-file-list contains set of files to move from repo into
users home directory. Also, it contains dirctories to create in
not available for use by vim and other programs. Most directories
are empty.

VomeroConsulting committed:
Added empty directories for vim use, needed to included .gitkeep

Commit:
added placeholder script that could be used with git

nJim commented on setup/homebrew.sh:
```diff
@@ -28,43 +38,67 @@ brew doctor
 brew install git
 
+brew install tkdiff
+k
```
Erroneous character?
